src/ui/MissingTerm/MissingTerms.ts

```diff
@@ -0,0 +1,218 @@
+     */
+    caption: ComponentOptions.buildLocalizedStringOption({ defaultValue: 'Missing' }),
+    /**
+     * The maximum number of missing term to be displayed
+     *
+     * **Default:** `5`
+     * **Minimum value:** `1`
+     */
+    numberOfResults: ComponentOptions.buildNumberOption({
+      defaultValue: 5,
+      min: 1
+    })
+  };
+
+  static doExport = () => {
+    exportGlobally({
+      MissingTerms: MissingTerms
+    });
+  };
+
+  /* Used to split terms and phrases. Match characters that can separate words or caracter for chinese, japanese and korean.
+  * Han: Unicode script for chinesse caracter
+  * We only need to import 1 asian script because what is important here is the space between the caracter and any script will contain it
+  */
+  private wordBoundary = '(([\\p{Han}])?([^(\\p{Latin}-)])|^|$)';
+  private termForcedToAppear: Array<string> = [];
+
+  /**
+   * Creates a new `MissingTerms` component instance.
+   * @param element The element on which to instantiate the component.
+   * @param options The configuration options for the component.
+   * @param bindings The bindings required by the component to function normally. If not set, these will be automatically resolved (with a slower execution time).
+   * @param result The query result item to associate the component with.
+   */
+  constructor(
+    public element: HTMLElement,
+    public options?: IMissingTermsOptions,
+    bindings?: IComponentBindings,
+    public result?: IQueryResult
+  ) {
+    super(element, MissingTerms.ID, bindings);
+
+    this.options = ComponentOptions.initComponentOptions(element, MissingTerms, options);
+    this.addMissingTerms();
+  }
+  /**
+   *Returns all original basic query expression terms that were not matched by the result item the component instance is associated with.
+   */
+  public get missingTerms(): string[] {
+    return this.result.absentTerms.filter(term => {
+      const regex = this.createWordBoundryDelimitedRegex(term);
+      return regex.test(this.queryStateModel.get('q'));
+    });
+  }
+
+  /**
+   * Inject a term in the advanced query.
+   */
+  public addTermForcedToAppear(term: string) {
+    if (this.missingTerms.indexOf(term) === -1) {
+      this.logger.warn(
+        'The term to re-inject is not present in the missing terms',
+        `You tried to inject "${term}" but the possible term to inject are: ${this.missingTerms.toString()}`,
+        'The query exited the function'
+      );
+      return;
+    }
+    this.updateTermForcedToAppear();
+    this.termForcedToAppear.push(term);
+    this.queryStateModel.set('missingTerm', [...this.termForcedToAppear]);
+  }
+
+  private updateTermForcedToAppear() {
+    this.termForcedToAppear = [...this.queryStateModel.get('missingTerm')];
+  }
+
+  private addMissingTerms() {
+    if (this.missingTerms.length === 0) {
+      return;
+    }
+    const missingTermelement = this.buildContainer().el;
```

Comment:
👍  for the camel-case

For the if it is necessary because I could have nothing return see line 130. I don't show wildcard word and if a featured result has a missing word, I hide that too.

What I will do to simplify a bit, il always return a result at line 130 but do the check from line 130 at line 117